tests/main/snap-disconnect/task.yaml

```
@@ -9,11 +9,24 @@ prepare: |
     echo "Install a test snap"
     snap pack "$TESTSLIB"/snaps/home-consumer
     snap install --dangerous "$SNAP_FILE"
+    snap install jq
 
 restore: |
     rm -f ./*.snap
 
 execute: |
+    #shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB"/systems.sh
+
+        #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+
+    inspect_connection() {
+      CONN="$1"
+      # shellcheck disable=SC2002
+      cat /var/lib/snapd/state.json | jq -r '.data["conns"] | has("'"$CONN"'")'
```

I think it's much safer to use `--arg CONN $CONN` with jq, I've struggled with jq filters and bash variables a lot before, maybe something like this (untested):

```suggestion
      cat /var/lib/snapd/state.json | jq --arg CONN $CONN -r '.data["conns"] | has($CONN)'
```